index.js

```diff
@@ -100,9 +100,37 @@ function _createConnection ({ connectionType, options, usePromise }, cb) {
   }
 }
 
+function isMySQLPoolOrPromisePool (obj) {
+  return obj && typeof obj.query === 'function' && typeof obj.execute === 'function' && typeof obj.getConnection === 'function' && typeof obj.pool === 'object'
+}
+
+function isMySQLConnectionOrPromiseConnection (obj) {
+  return obj && typeof obj.query === 'function' && typeof obj.execute === 'function' && typeof obj.connection === 'object'
+}
+
+function isMySQLPool (obj) {
+  return isMySQLPoolOrPromisePool(obj) && typeof obj.pool.promise === 'function'
+}
+
+function isMySQLPromisePool (obj) {
+  return isMySQLPoolOrPromisePool(obj) && typeof obj.threadId === 'number' && obj.pool.promise === undefined
```

Comment:
Where `threadId` is defined? To be honest, I didn't found this in the `mysql2` code, and I think it's why the workflow doesn't work.
```suggestion
  return isMySQLPoolOrPromisePool(obj) && obj.pool.promise === undefined
```

Reply:
`threadId` is defined on the `Connection` type in `promise.d.ts` [here](https://github.com/sidorares/node-mysql2/blob/master/promise.d.ts#L45).
Since `Pool` extends `Connection` [here](https://github.com/sidorares/node-mysql2/blob/master/promise.d.ts#L84) should be available to both promise objects

Reply:
If you look closer at their code, the [promise pool](https://github.com/sidorares/node-mysql2/blob/master/promise.js#L329) does not extends the `Connection` class, but the [`PromisePoolConnection`](https://github.com/sidorares/node-mysql2/blob/master/promise.js#L316) does.
So the only way to get it is to get the connection from the pool.
```js
const connection = await obj.pool.getConnection().catch(() => null)
connection.threadId
```

Reply:
Interesting I am going by mysql promise types [file](https://github.com/sidorares/node-mysql2/blob/master/promise.d.ts#L84). Which has it defined
![Screenshot 2024-03-29 at 8 01 12 AM](https://github.com/fastify/fastify-mysql/assets/15673208/41a03681-77b2-4bd1-9fb5-05af7eca5763)